date-time-picker/src/iosMain/kotlin/DateTimePickerPresenter.kt

```diff
@@ -91,33 +105,66 @@ actual class DateTimePickerPresenter(
-            datePickerView.bottomAnchor.constraintEqualToAnchor(view.bottomAnchor, -120.0 as CGFloat).active = true
-            datePickerView.widthAnchor.constraintEqualToConstant(view.frame.useContents { (size.width - 20.0) as CGFloat }).active = true
-            datePickerView.centerXAnchor.constraintEqualToAnchor(view.centerXAnchor).active = true
-            view.translatesAutoresizingMaskIntoConstraints = false
-        }.run {
-            parent.presentViewController(this, animated, null)
+
+            val topAnchor = datePicker.message?.let {
+                UILabel().apply {
+                    text = it
+                    containerView.addSubview(this)
+                    translatesAutoresizingMaskIntoConstraints = false
+                    bottomAnchor.constraintEqualToAnchor(
+                        datePickerView.topAnchor,
+                        -15.0 as CGFloat
+                    ).active = true
+                    containerView.leadingAnchor.constraintEqualToAnchor(
+                        leadingAnchor,
+                        -20.0 as CGFloat
+                    ).active = true
+                    containerView.trailingAnchor.constraintEqualToAnchor(
+                        trailingAnchor,
+                        20.0 as CGFloat
+                    ).active = true
+                    bottomAnchor
+                }.topAnchor
+            } ?: datePickerView.topAnchor
+
+            topAnchor.constraintEqualToAnchor(containerView.topAnchor, 10.0 as CGFloat).active = true
+
+            UIButton().apply {
+                addTarget(this@DateTimePickerViewController, sel_registerName("onSelected"), UIControlEventTouchUpInside)
+                view.insertSubview(this, 0)
+                translatesAutoresizingMaskIntoConstraints = false
+                view.topAnchor.constraintEqualToAnchor(this.topAnchor).active = true
+                view.bottomAnchor.constraintEqualToAnchor(bottomAnchor).active = true
+                view.leadingAnchor.constraintEqualToAnchor(leadingAnchor).active = true
+                view.trailingAnchor.constraintEqualToAnchor(trailingAnchor).active = true
+            }
+        }
+
+        @ObjCAction
+        private fun onSelected() {
+            completion(Date.epoch((datePickerView.date.timeIntervalSince1970 * 1000.0).toLong(), datePicker.selectedDate.timeZone, datePicker.locale))
+            dismissDateTimePicker(true)
         }
     }
+
+    actual class Builder(private val viewController: UIViewController) : BaseDateTimePickerPresenter.Builder() {
+        actual override fun create(coroutineScope: CoroutineScope) =
+            DateTimePickerPresenter(createDateTimePicker(), viewController)
+    }
+
+    override fun dismissDateTimePicker(animated: Boolean) {
+        parent.dismissModalViewControllerAnimated(animated)
+    }
+
+    override fun showDateTimePicker(animated: Boolean, completion: (Date?) -> Unit) {
+        dismissDateTimePicker(animated)
```

Comment:
Once use dismiss and present animated in same time can use ensure here is not conflicts like presenting view when view already presented?

Reply:
@avdyushin could you rephrase that?

Reply:
You can see `Warning: Attempt to present * on * which is already presenting (null)` and view is not presented in some edge cases